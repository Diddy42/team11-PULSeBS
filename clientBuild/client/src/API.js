async function userLogin(username, password) {
    return new Promise((resolve, reject) => {
        fetch('http://localhost/server/src/api/Login.php', {
            method: 'POST',
            headers: {
                //'Content-Type': 'application/json',
                'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
            },
            body: JSON.stringify({ username: username, password: password }),
        }).then((response) => {
            response.json().then((obj) => {
                /*
                if(obj === 0 ) {        //Da modificare in base a quanto ritorna il server. Sarebbe utile che ritorni l'id o anche nome e cognome la visualizzare, e il ruolo
                    reject(obj);
                } else {
                    resolve(obj);    
                }
                */
                resolve(obj);
            })
                .catch((err) => { reject({ errors: [{ param: "Application", msg: "Cannot parse server response" }] }) });
        }).catch((err) => { reject({ errors: [{ param: "Server", msg: "Cannot communicate" }] }) });
    });
}

async function logout() {
    return new Promise((resolve, reject) => {
        fetch('http://localhost:80/server/api/logout.php')      //Da verificare
            .then((response) => {
                if (response.ok) {
                    resolve(0);
                }
                else {
                    reject();
                }
            })
            .catch((err) => {
                reject(err);
            });
    });
}

//Ritorna le lezioni prenotabili dallo studente
async function getBookableStudentLectures(studentId) {
    const url = "http://localhost/server/bookableLessons/" + studentId;
    const response = await fetch(url);
    const lectures = await response.json();
    if (response.ok) {
        if (lectures === 0) {
            return [];
        }
        return lectures;
    } else {
        let err = { status: response.status, errorObj: lectures };
        throw err;
    }
}

//Ritorna le prenotazioni effettuate dagli studenti per la lezione indicata (teacher side)
async function getBooking(lectureId) {
    const url = "http://localhost/server/bookedStudentsForLecture/" + lectureId;
    const response = await fetch(url);
    const booking = await response.json();
    if (response.ok) {
        if (booking === 0) {
            return [];
        }
        return booking;
    } else {
        let err = { status: response.status, errorObj: booking };
        throw err;
    }
}

// API to retrieve all the bookings of a student
async function getStudentBookings(studentId) {
    const url = "http://localhost/server/studentBookings/" + studentId;
    const response = await fetch(url);
    const booking = await response.json();
    if (response.ok) {
        if (booking === 0) {
            return [];
        }
        return booking;
    } else {
        let err = { status: response.status, errorObj: booking };
        throw err;
    }
}

//API per prenotare un posto a lezione
async function bookASeat(lectureId, studentId, date) {
    const url = "http://localhost/server/insertLecture"
    const response = await fetch(url, {
        method: 'POST',
        headers: {
            //'Content-Type': 'application/json',
            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
        },
        body: JSON.stringify({
            idLesson: lectureId,
            idUser: studentId,
            date: date
        }),
    });

    try {
        const resJ = await response.json();
        if (response.ok) {
            return resJ;
        } else {
            let err = { status: response.status, errorObj: resJ };
            throw err;
        }
    }
    catch (e) {
        console.log("Error in booking a seat: " + e);
        return 0;   //meaning: no new customer
    }
}

// API to retrieve all the lectures of a teacher
async function getTeacherLectures(teacherId) {
    const url = "http://localhost/server/scheduledLecturesForTeacher/" + teacherId
    const response = await fetch(url);
    const lectures = await response.json();
    if (response.ok) {
        if (lectures === 0) {
            return [];
        }
        return lectures;
    } else {
        let err = { status: response.status, errorObj: lectures };
        throw err;
    }
}

async function deleteBooking(bookingId) {
    const url = "http://localhost/server/updateBooking/" + bookingId;
    return new Promise((resolve, reject) => {
        fetch(url, {
            method: 'PUT',
            headers: {
                //'Content-Type': 'application/json',
                'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
            },
            body: JSON.stringify({
            }),
        }).then((response) => {
            if (response.ok) {
                resolve(0);
            } else {
                reject({ errore: "Error in deleting a booking" });
            }
        }).catch((err) => { reject({ errors: [{ param: "Server", msg: "Cannot communicate" }] }) });
    });
}

async function deleteLecture(lectureId) {
    const url = "http://localhost/server/deleteLecture/" + lectureId;
    const response = await fetch(url, {
        method: 'PUT',
        headers: {
            //'Content-Type': 'application/json',
            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
        },
        body: JSON.stringify({
        }),
    });
    const result = await response.json();
    if (response.ok) {
        if (result === 1) {
            return 1;
        } else {
            console.log("Too late to delete the lecture");
            return -1;
        }
    } else {
        let err = { status: response.status, errorObj: result };
        throw err;
    }
}

async function changeToOnline(lectureId) {
    const url = "http://localhost/server/changeToOnline/" + lectureId;
    const response = await fetch(url, {
        method: 'PUT',
        headers: {
            //'Content-Type': 'application/json',
            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
        },
        body: JSON.stringify({
        }),
    });
    const result = await response.json();
    if (response.ok) {
        if (result === 1) {
            return 1;
        } else {
            console.log("Too late to change the lecture");
            return -1;
        }
    } else {
        let err = { status: response.status, errorObj: result };
        throw err;
    }
}

async function getAllCourses() {
    const url = "http://localhost/server/courses";

    const response = await fetch(url);
    const courses = await response.json();
    if (response.ok) {
        if (courses === 0) {
            return [];
        }
        return courses;
    } else {
        let err = { status: response.status, errorObj: courses };
        throw err;
    }
}

const ALL_COURSES_FILTER = "L.idCourse";
async function getBookingStatisticsByMonth(courseIds) {
    const url = `http://localhost/server/bookingStatistics/?filterTime=year,monthOfYear&filterCourse=${courseIds.toString()}`;

    const response = await fetch(url);
    const stats = await response.json();
    if (response.ok) {
        if (stats === 0) {
            return [];
        }
        return stats;
    } else {
        let err = { status: response.status, errorObj: stats };
        throw err;
    }
} 
async function getBookingStatisticsByWeek(courseIds) {
    const url = `http://localhost/server/bookingStatistics/?filterTime=year_month_week&filterCourse=${courseIds.toString()}`;

    const response = await fetch(url);
    const stats = await response.json();
    if (response.ok) {
        if (stats === 0) {
            return [];
        }
        return stats;
    } else {
        let err = { status: response.status, errorObj: stats };
        throw err;
    }
}
async function getBookingStatisticsByLesson(courseIds) {
    const url = `http://localhost/server/bookingStatistics/?filterTime=L.idLesson&filterCourse=${courseIds.toString()}`;

    const response = await fetch(url);
    const stats = await response.json();
    if (response.ok) {
        if (stats === 0) {
            return [];
        }
        return stats;
    } else {
        let err = { status: response.status, errorObj: stats };
        throw err;
    }
}

async function getCoursesOfTeacher(teacherId) {
    const url = "http://localhost/server/teacherCourses/" + teacherId;

    const response = await fetch(url);
    const courses = await response.json();
    if (response.ok) {
        if (courses === 0) {
            return [];
        }
        return courses;
    } else {
        let err = { status: response.status, errorObj: courses };
        throw err;
    }
}

const API = {
    userLogin, logout, getBookableStudentLectures, getBooking, getStudentBookings, bookASeat, deleteBooking, getTeacherLectures, deleteLecture, changeToOnline, getAllCourses, 
    getBookingStatisticsByMonth, getBookingStatisticsByWeek, getBookingStatisticsByLesson, ALL_COURSES_FILTER, getCoursesOfTeacher
};
export default API;
