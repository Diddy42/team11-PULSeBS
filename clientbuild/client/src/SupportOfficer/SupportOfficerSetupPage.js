import React from "react";
import { Stepper, Step, StepLabel, StepContent, Button, Typography } from '@material-ui/core';
import Dropzone from "./Dropzone"
import Papa from 'papaparse';
import API from '../API';

class SupportOfficerSetupPage extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            files: [],
            activeStep: 0,
            steps: [
                'Students', 'Teachers', 'Courses', 'Lectures', 'Classes', 'Submit'
            ]
        };

        this.handleNext = this.handleNext.bind(this);
        this.handleBack = this.handleBack.bind(this);
        this.handleReset = this.handleReset.bind(this);

        this.onFileAdded = this.onFileAdded.bind(this);
    }

    componentDidMount = () => {
    }

    handleNext() {
        this.setState(prevState => ({
            activeStep: prevState.activeStep + 1
        }), () => {
            //callback after successful state update
            Papa.parse(this.state.files[0], {
                complete: (res) => {
                    console.log(res.data);

                    API.setUpStudents(res.data)
                    .then((res) => {
                        console.log('set up students successful');
                    })
                    .catch((err) => {
                        console.log('error in setting up students');
                        console.log(err);
                    });
                }
            });
        });
    };

    handleBack() {
        this.setState(prevState => ({
            activeStep: prevState.activeStep - 1
        }));
    };

    handleReset() {
        this.setState({
            activeStep: 0
        });
    };

    getStepContent(stepIndex) {
        switch (stepIndex) {
            case 0:
                return `upload the students csv file.`;
            case 1:
                return `upload the teachers csv file.`;
            case 2:
                return `upload the courses csv file.`;
            case 3:
                return `upload the lectures csv file.`;
            case 4:
                return `upload the classes csv file.`;
            case 5:
                return `check the files and submit.`;
            default:
                return <div>Unknown stepIndex</div>;
        }
    }

    onFileAdded(file) {
        const activeStep = this.state.activeStep;
        /* replace or add the file */
        this.setState(prevState=> {
            for(let f of prevState.files) {
                /* check the new file is not present in files */ 
                if(f !== undefined && file.name === f.name) return ({});
            }
            prevState.files[activeStep] = file;
            return ({ files: prevState.files });
        })
    }

    render() {
        return <>
            <div className="d-flex align-items-center" style={{ height: "calc(100% - 56px" }}>
                <Stepper style={{ minWidth: "350px", marginLeft:"64px", marginRight:"128px" }} activeStep={this.state.activeStep} orientation="vertical">
                    {this.state.steps.map((label, index) => (
                        <Step key={label}>
                            <StepLabel>{label}</StepLabel>
                            <StepContent>
                                <Typography>{this.getStepContent(index)}</Typography>
                                <div className="my-3">
                                    <div>
                                        <Button
                                            disabled={this.state.activeStep === 0}
                                            onClick={this.handleBack}
                                            style={{ marginRight:"8px"}}
                                        >
                                            Back
                                        </Button>
                                        <Button
                                            variant="contained"
                                            color="primary"
                                            disabled={this.state.files[this.state.activeStep] === undefined && this.state.files.length !== 5}
                                            onClick={this.handleNext}
                                            style={{ marginLeft:"8px"}}
                                        >
                                            {this.state.activeStep === this.state.steps.length - 1 ? 'Finish' : 'Next'}
                                        </Button>
                                    </div>
                                </div>
                            </StepContent>
                        </Step>
                    ))}
                </Stepper>

                <div className="w-100 d-flex align-items-center h-100" style={{ marginLeft:"128px", marginRigth:"64px" }}>
                    <div>
                        <Dropzone
                            onFileAdded={this.onFileAdded}
                            disabled={this.state.activeStep === this.state.steps.length-1}
                            file={this.state.files[this.state.activeStep]}
                        />
                    </div>
                    <div style={{ marginLeft:"32px" }}>
                        {this.state.files.map(file => {
                            return (
                                <div key={file.name} style={{ height: "50px", padding: "8px", overflow: "hidden" }}>
                                    <span style={{ marginBottom: "8px", fontSize: "16px", color: "#555", textOverflow: "ellipsis", overflow: "hidden" }}>{file.name}</span>
                                </div>
                            );
                        })}
                    </div>

                </div>
            </div>
        </>
    }

}

export default SupportOfficerSetupPage;
