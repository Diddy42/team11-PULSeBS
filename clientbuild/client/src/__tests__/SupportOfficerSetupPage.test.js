import TestRenderer from 'react-test-renderer';
import React from 'react';
import SupportOfficerSetupPage from '../SupportOfficer/SupportOfficerSetupPage';
import API from '../API';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { fireEvent } from '@testing-library/react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Papa from 'papaparse';

configure({ adapter: new Adapter() });

test('renders SupportOfficerPage page', () => {
    const tree = TestRenderer.create(<SupportOfficerSetupPage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

test("handleNext", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    instance.handleFinish = jest.fn();
    expect(instance.state.activeStep).toBe(0);
    await instance.handleNext();
    expect(instance.state.activeStep).toBe(1);
    await instance.handleNext();
    await instance.handleNext();
    await instance.handleNext();
    await instance.handleNext();
    await instance.handleNext();
    expect(instance.state.activeStep).toBe(6);
    expect(instance.handleFinish).toHaveBeenCalledTimes(1);
});

test("handleBack and handleReset", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    expect(instance.state.activeStep).toBe(0);
    await instance.handleNext();
    expect(instance.state.activeStep).toBe(1);
    await instance.handleBack();
    expect(instance.state.activeStep).toBe(0);
    await instance.handleNext();
    await instance.handleNext();
    expect(instance.state.activeStep).toBe(2);
    await instance.handleReset();
    expect(instance.state.activeStep).toBe(0);
});

test("getStepContent", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    let res = instance.getStepContent(0);
    expect(res).toBe(`upload the students csv file.`);
    res = instance.getStepContent(1);
    expect(res).toBe(`upload the professors csv file.`);
    res = instance.getStepContent(2);
    expect(res).toBe(`upload the courses csv file.`);
    res = instance.getStepContent(3);
    expect(res).toBe(`upload the schedule csv file.`);
    res = instance.getStepContent(4);
    expect(res).toBe(`upload the enrollment csv file.`);
    res = instance.getStepContent(5);
    expect(res).toBe(`check the files and submit.`);
    res = instance.getStepContent(6);
    expect(res.props.children).toBe("Unknown stepIndex");
});

test("onFileAdded", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    await instance.onFileAdded({name: "testFile"});
    expect(instance.state.files[0].name).toBe("testFile");
    expect(instance.state.files.length).toBe(1);
    await instance.handleNext();
    await instance.onFileAdded({name: "testFile"});
    expect(instance.state.files[0].name).toBe("testFile");
    expect(instance.state.files.length).toBe(1);
    await instance.onFileAdded({name: "testFile2"});
    expect(instance.state.files.length).toBe(2);
});

test("csvDataToJSON", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    const csvTest = [["field1", "field2"], ["data11", "data12"], ["data21", "data22"]];
    const expectedRes = [{field1: "data11", field2: "data12"}, {field1: "data21", field2: "data22"}];
    const res = await instance.csvDataToJSON(csvTest);
    expect(res).toStrictEqual(expectedRes);
});

test("handleFinish", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    instance.convertAndSendClasses = jest.fn();
    instance.convertAndSendStudents = jest.fn();
    instance.convertAndSendTeachers = jest.fn();
    instance.convertAndSendLectures = jest.fn();
    instance.convertAndSendCourses = jest.fn();
    await instance.handleFinish();
    expect(instance.convertAndSendClasses).toHaveBeenCalledTimes(1);
    expect(instance.convertAndSendStudents).toHaveBeenCalledTimes(1);
    expect(instance.convertAndSendTeachers).toHaveBeenCalledTimes(1);
    expect(instance.convertAndSendCourses).toHaveBeenCalledTimes(1);
    expect(instance.convertAndSendLectures).toHaveBeenCalledTimes(1);
    expect(instance.state.showResultModal).toBe(true);
    expect(instance.state.setupResult).toBe(true);
});

test("handleFinish error", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    //API.setUpProfessors = jest.fn(() => Promise.reject([]));
    instance.convertAndSendClasses = jest.fn(() => Promise.reject([]));
    instance.convertAndSendStudents = jest.fn();
    instance.convertAndSendTeachers = jest.fn();
    instance.convertAndSendLectures = jest.fn();
    instance.convertAndSendCourses = jest.fn();
    await instance.handleFinish();
    expect(instance.state.showResultModal).toBe(true);
    expect(instance.state.setupResult).toBe(false);
    expect(instance.convertAndSendStudents).toHaveBeenCalledTimes(1);
    expect(instance.convertAndSendTeachers).toHaveBeenCalledTimes(1);
    expect(instance.convertAndSendCourses).toHaveBeenCalledTimes(1);
    expect(instance.convertAndSendLectures).toHaveBeenCalledTimes(1);
});

test("convertAndSendTeachers", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    API.setUpProfessors = jest.fn(() => Promise.resolve([]));
    const blob=new Blob(['fieldA,fieldB,fieldC'],{type:'text/csv'});
    Object.assign(blob,{name:'dummy.txt'});
    Object.assign(blob,{lastModifiedDate :new Date()});
    const file = new File([blob], 'dummy.txt', { type: 'text/csv' });
    const res = await instance.convertAndSendTeachers(file);
    API.setUpProfessors = jest.fn(() => Promise.reject([]));
    try {
        const res2 = await instance.convertAndSendTeachers(file);
        expect(true).toBe(false);
    } catch(e) {
        expect(true).toBe(true);
    }
});

test("convertAndSendCourses", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    API.setUpCourses = jest.fn(() => Promise.resolve([]));
    const blob=new Blob(['fieldA,fieldB,fieldC'],{type:'text/csv'})
    Object.assign(blob,{name:'dummy.txt'})
    Object.assign(blob,{lastModifiedDate :new Date()})
    const file = new File([blob], 'dummy.txt', { type: 'text/csv' })
    const res = await instance.convertAndSendCourses(file);
    API.setUpCourses = jest.fn(() => Promise.reject([]));
    try {
        const res2 = await instance.convertAndSendCourses(file);
        expect(true).toBe(false);
    } catch(e) {
        expect(true).toBe(true);
    }
});

test("convertAndSendLectures", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    API.setUpLectures = jest.fn(() => Promise.resolve([]));
    const blob=new Blob(['fieldA,fieldB,fieldC'],{type:'text/csv'});
    Object.assign(blob,{name:'dummy.txt'});
    Object.assign(blob,{lastModifiedDate :new Date()});
    const file = new File([blob], 'dummy.txt', { type: 'text/csv' });
    const res = await instance.convertAndSendLectures(file);
    API.setUpLectures = jest.fn(() => Promise.reject([]));
    try {
        const res2 = await instance.convertAndSendLectures(file);
        expect(true).toBe(false);
    } catch(e) {
        expect(true).toBe(true);
    }
});

test("convertAndSendStudents", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    API.setUpStudents = jest.fn(() => Promise.resolve([]));
    const blob=new Blob(['fieldA,fieldB,fieldC'],{type:'text/csv'});
    Object.assign(blob,{name:'dummy.txt'});
    Object.assign(blob,{lastModifiedDate :new Date()});
    const file = new File([blob], 'dummy.txt', { type: 'text/csv' });
    const res = await instance.convertAndSendStudents(file);
    API.setUpStudents = jest.fn(() => Promise.reject([]));
    try {
        const res2 = await instance.convertAndSendStudents(file);
        expect(true).toBe(false);
    } catch(e) {
        expect(true).toBe(true);
    }
});

test("convertAndSendClasses", async () => {
    const component = await shallow(<SupportOfficerSetupPage/>);
    const instance = await component.instance();
    API.setUpClasses = jest.fn(() => Promise.resolve([]));
    const blob=new Blob(['fieldA,fieldB,fieldC'],{type:'text/csv'});
    Object.assign(blob,{name:'dummy.txt'});
    Object.assign(blob,{lastModifiedDate :new Date()});
    const file = new File([blob], 'dummy.txt', { type: 'text/csv' });
    const res = await instance.convertAndSendClasses(file);
    API.setUpClasses = jest.fn(() => Promise.reject([]));
    try {
        const res2 = await instance.convertAndSendClasses(file);
        expect(true).toBe(false);
    } catch(e) {
        expect(true).toBe(true);
    }
});