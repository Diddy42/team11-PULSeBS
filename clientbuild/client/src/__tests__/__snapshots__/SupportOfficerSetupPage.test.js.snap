// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SupportOfficerPage page 1`] = `
<div
  className="d-flex align-items-center"
  style={
    Object {
      "height": "calc(100% - 56px",
    }
  }
>
  <div
    className="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
    style={
      Object {
        "marginLeft": "64px",
        "marginRight": "128px",
        "minWidth": "350px",
      }
    }
  >
    <div
      className="MuiStep-root MuiStep-vertical"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              1
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            Students
          </span>
        </span>
      </span>
      <div
        className="MuiStepContent-root"
        disabled={false}
        icon={1}
      >
        <div
          className="MuiCollapse-container MuiStepContent-transition MuiCollapse-entered"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                upload the students csv file.
              </p>
              <div
                className="my-3"
              >
                <div>
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
                    disabled={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "marginRight": "8px",
                      }
                    }
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Back
                    </span>
                  </button>
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
                    disabled={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "marginLeft": "8px",
                      }
                    }
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Next
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical Mui-disabled"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      className="MuiStep-root MuiStep-vertical"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Teachers
          </span>
        </span>
      </span>
      <div
        className="MuiStepContent-root"
        disabled={true}
        icon={2}
      />
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical Mui-disabled"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      className="MuiStep-root MuiStep-vertical"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Courses
          </span>
        </span>
      </span>
      <div
        className="MuiStepContent-root"
        disabled={true}
        icon={3}
      />
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical Mui-disabled"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      className="MuiStep-root MuiStep-vertical"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              4
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Lectures
          </span>
        </span>
      </span>
      <div
        className="MuiStepContent-root"
        disabled={true}
        icon={4}
      />
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical Mui-disabled"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      className="MuiStep-root MuiStep-vertical"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              5
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Classes
          </span>
        </span>
      </span>
      <div
        className="MuiStepContent-root"
        disabled={true}
        icon={5}
      />
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical Mui-disabled"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      className="MuiStep-root MuiStep-vertical"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              6
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Submit
          </span>
        </span>
      </span>
      <div
        className="MuiStepContent-root MuiStepContent-last"
        disabled={true}
        icon={6}
      />
    </div>
  </div>
  <div
    className="w-100 d-flex align-items-center h-100"
    style={
      Object {
        "marginLeft": "128px",
        "marginRigth": "64px",
      }
    }
  >
    <div>
      <div
        className="Dropzone "
        onClick={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <input
          accept=".csv"
          className="FileInput"
          onChange={[Function]}
          type="file"
        />
        <svg
          className="UploadIcon my-3"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z"
          />
        </svg>
        <span
          className="Filename"
        >
          Choose a file
        </span>
      </div>
    </div>
    <div
      style={
        Object {
          "marginLeft": "32px",
        }
      }
    />
  </div>
</div>
`;
